GeoFire/Implementation/GFQuery.m

```diff
@@ -175,6 +176,9 @@ @interface GFQuery ()
 @property (nonatomic, strong) NSMutableDictionary *firebaseHandles;
 @property (nonatomic, strong) NSMutableSet *outstandingQueries;
 
+@property (nonatomic, strong) NSMutableDictionary *withSnapshotKeyEnteredObservers;
+@property (nonatomic, strong) NSMutableDictionary *withSnapshotKeyExitedObservers;
+@property (nonatomic, strong) NSMutableDictionary *withSnapshotKeyMovedObservers;
```

Comment:
Consider adding a comment to briefly explain the difference between these snapshots and the below observers.